src/btree/bt_curprev.c

```diff
@@ -630,6 +631,13 @@ __cursor_row_prev(
             key->data = WT_INSERT_KEY(ins);
             key->size = WT_INSERT_KEY_SIZE(ins);
 
+            if (WT_IS_HS(session->dhandle) && F_ISSET(&cbt->iface, WT_CURSTD_BOUND_UPPER)) {
```

Comment:
I understand why this is needed but I can't say I'm a fan of it. I think we'll need to think of a more generalizable solution.

Reply:
I'm not either. Do you have any better idea?

Reply:
At this stage my only thought is wondering if we can go a layer down and write a __cursor_row_search_before / after that intentionally places the cursor at the lowest level of WiredTiger. That sounds tedious and not like what we'd want right now. At the very least we should move this into its own function, I'll keep thinking, and have looped in Jie.